src/resources/js/ian-client.js

```diff
@@ -82,15 +89,56 @@
 					if (!e.composedPath().includes(Ian.sidebar) && !e.composedPath().includes(Ian.icon)) {
 						Ian.sidebar.classList.add("is-hidden");
 					}
+					break;
 			}
 		};
 
 		const handleKeydown = e => {
 			if (["Escape", "Esc"].includes(e.key) || e.keyCode === 27) {
 				Ian.sidebar.classList.add("is-hidden");
+				calculateSidebarPosition();
 			}
 		};
 
+		const calculateSidebarPosition = () => {
+			const bar = document.getElementById("wpadminbar");
+			const wrapper = document.querySelector(".wrap .ian-header");
+			if (wrapper) {
+				const rect = wrapper.getBoundingClientRect();
```

Comment:
Is wrapper or settingstabs ever on the same page? 

The way it is setup settingstabs will always set the positions if on the same page, is that correct?

Reply:
Hi @jesseeproductions , either will exist, not both. The one is the case in our Settings pages and the other is the case in the native Events Table or Edit Events pages.